tests/__init__.py

```diff
@@ -457,7 +457,7 @@ def test_modified_uids(self):
                 (
                     "bash",
                     "-xc",
-                    'test "$(stat -c \'%U:%G\' /opt/odoo/custom/src)" == "root:odoo"',
+                    'test "$(stat -c \'%U:%G\' /opt/odoo/custom/src)" == "odoo:odoo"',
```

Comment:
This is a safety belt against malicious code: odoo and addons code is unmodifiable by the odoo user itself.

Sorry but this must stay as it was... 🙄

Reply:
ok, will change that, i was not sure because /opt/odoo/custom/src was chowned to odoo:odoo in `autoaggregate` and then chowned to root:odoo in 800-permissions

how do root:odoo permissions work with dev mode (/opt/odoo/custom mapped to the users working directory)?

Reply:
> how do root:odoo permissions work with dev mode (/opt/odoo/custom mapped to the users working directory)?

The chown to `root:odoo` is done here: https://github.com/Tecnativa/doodba/blob/1c8d05aab51dfd21ed8a55af7ea30302cff614a9/build.d/800-permissions#L3

That's at build time.

However, in devel, your code is bind-mounted into the container, so whatever was in that directory at build time is overriden, and you can write to it.

Besides, [`invoke` tasks in the template automatically set UID, GID and UMASK](https://github.com/Tecnativa/doodba-copier-template/blob/72b97c1bafa69b9474ff948f08914e12d3a326f3/tasks_downstream.py#L21) to make `autoaggregate` behave as you would, outside the container, to avoid developer friction

Reply:
Ah yes, that's right, somehow i didn't realize the difference between build and runtime mappings. Thanks for the explanation.